pints/_transformation.py

```diff
@@ -0,0 +1,574 @@
+#
+# Transformation functions
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+from scipy.special import logit, expit
+
+
+class Transform(object):
+    """
+    Abstract base class for objects that provide some convenience parameter
+    transformation from the model parameter space to a search space.
+
+    If ``t`` is an instance of a ``Transform`` class, you can apply
+    the transformation of a parameter vector from the model space ``p`` to the
+    search space ``x`` by using ``x = t.to_search(p)`` and the inverse by using
+    ``p = t.to_model(x)``.
+    """
+    def apply_log_pdf(self, log_pdf):
+        """
+        Returns a transformed log-PDF class.
+        """
+        return TransformedLogPDF(log_pdf, self)
+
+    def apply_error_measure(self, error_measure):
+        """
+        Returns a transformed error measure class.
+        """
+        return TransformedErrorMeasure(error_measure, self)
+
+    def apply_boundaries(self, boundaries):
+        """
+        Returns a transformed boundaries class.
+        """
+        return TransformedBoundaries(boundaries, self)
+
+    def jacobian(self, x):
+        """
+        Returns the Jacobian for a parameter vector ``x`` in the search space.
+
+        *This is an optional method. It is needed when `sigma0` is provided in
```

Comment:
As stated earlier, might be better to have a general covariance matrix transformation method!

(Even if we don't would update this to say it can be used for any old sigma, not just sigma0 in a controller!)

Reply:
Done.